src/components/flat-table/flat-table-row/flat-table-row.style.ts

```diff
@@ -237,7 +261,15 @@ const StyledFlatTableRow = styled.tr<StyledFlatTableRowProps>`
       ${StyledFlatTableHeader} {
         border-bottom: 1px solid ${borderColor(colorTheme)};
 
+        ${horizontalBorderSize !== "small" &&
+        css`
+          border-bottom: ${horizontalBorderSizes[horizontalBorderSize]} solid
+            var(--colorsUtilityMajor100);
+        `}
+
         ${!isInSidebar &&
+        colorTheme !== "transparent-white" &&
```

Comment:
suggestion (non-blocking):
```suggestion
        !colorTheme.includes("transparent") && `
```